core/src/main/java/org/apache/spark/shuffle/sort/io/DefaultShuffleMapOutputWriter.java

```diff
@@ -222,18 +223,10 @@ public void write(byte[] buf, int pos, int length) throws IOException {
     }
 
     @Override
-    public void close() throws IOException {
-      flush();
```

Comment:
wait, why did we get rid of this?

Reply:
On one hand it's inconsequential because the contract is the writer above this never closes this stream - it leaves it up to the partition writer to close this.

On the other hand, this is to preserve an optimization we discovered in `UnsafeShuffleWriter` that encourages us to minimize the number of times we flush the buffer to the underlying file.

We can get correctness by just flushing the merged buffered file stream at the very end when we commit all the partitions.